src/pudl/metadata/classes.py

```diff
@@ -139,151 +163,36 @@ def _get_jinja_environment(template_dir: DirectoryPath = None):
-
-    Examples:
-        >>> class Class(Base):
-        ...     fields_: list[str] = pydantic.Field(alias="fields")
-        >>> m = Class(fields=['x'])
-        >>> m
-        Class(fields=['x'])
-        >>> m.fields
-        ['x']
-        >>> m.fields = ['y']
-        >>> m.dict()
-        {'fields': ['y']}
-    """
-
-    class Config:
-        """Custom Pydantic configuration."""
-
-        validate_all: bool = True
-        validate_assignment: bool = True
-        extra: str = "forbid"
-        arbitrary_types_allowed = True
-
-    def dict(self, *args, by_alias=True, **kwargs) -> dict:  # noqa: A003
-        """Return as a dictionary."""
-        return super().dict(*args, by_alias=by_alias, **kwargs)
-
-    def json(self, *args, by_alias=True, **kwargs) -> str:
-        """Return as JSON."""
-        return super().json(*args, by_alias=by_alias, **kwargs)
-
-    def __getattribute__(self, name: str) -> Any:
-        """Get attribute."""
-        if name in ("fields", "schema") and f"{name}_" in self.__dict__:
-            name = f"{name}_"
-        return super().__getattribute__(name)
-
-    def __setattr__(self, name, value) -> None:
-        """Set attribute."""
-        if name in ("fields", "schema") and f"{name}_" in self.__dict__:
-            name = f"{name}_"
-        super().__setattr__(name, value)
-
-    def __repr_args__(self) -> list[tuple[str, Any]]:
-        """Returns the attributes to show in __str__, __repr__, and __pretty__."""
-        return [
-            (a[:-1] if a in ("fields_", "schema_") else a, v)
-            for a, v in self.__dict__.items()
-        ]
-
-
 # ---- Class attribute types ---- #
 
 # NOTE: Using regex=r"^\S(.*\S)*$" to fail on whitespace is too slow
-String = pydantic.constr(min_length=1, strict=True, regex=r"^\S+(\s+\S+)*$")
+String = Annotated[
+    str, StringConstraints(min_length=1, strict=True, pattern=r"^\S+(\s+\S+)*$")
+]
 """Non-empty :class:`str` with no trailing or leading whitespace."""
 
-SnakeCase = pydantic.constr(
-    min_length=1, strict=True, regex=r"^[a-z_][a-z0-9_]*(_[a-z0-9]+)*$"
-)
+SnakeCase = Annotated[
+    str,
+    StringConstraints(
+        min_length=1, strict=True, pattern=r"^[a-z_][a-z0-9_]*(_[a-z0-9]+)*$"
+    ),
+]
 """Snake-case variable name :class:`str` (e.g. 'pudl', 'entity_eia860')."""
 
-Bool = pydantic.StrictBool
-"""Any :class:`bool` (`True` or `False`)."""
-
-Float = pydantic.StrictFloat
-"""Any :class:`float`."""
-
-Int = pydantic.StrictInt
-"""Any :class:`int`."""
-
-PositiveInt = pydantic.conint(ge=0, strict=True)
+PositiveInt = Annotated[int, pydantic.Field(ge=0, strict=True)]
 """Positive :class:`int`."""
 
-PositiveFloat = pydantic.confloat(ge=0, strict=True)
+PositiveFloat = Annotated[float, pydantic.Field(ge=0, strict=True)]
 """Positive :class:`float`."""
 
-Email = pydantic.EmailStr
-"""String representing an email."""
-
-HttpUrl = pydantic.AnyHttpUrl
-"""Http(s) URL."""
-
-
```

Comment:
It seemed more straightforward to me to just use the pydantic types directly in the classes where these types are used, rather than renaming them and creating a layer of indirection.